src/nats.h

```diff
@@ -7104,6 +7104,1187 @@ kvStatus_Destroy(kvStatus *sts);
+
+/** @} */ // end of microTypes
+
+/** \defgroup microCallbacks Callbacks
+ *
+ *  Microservice callbacks.
+ *  @{
+ */
+
+/** 
+ * @brief Callback type for request processing.
+ *
+ * This is the callback that one provides when creating a microservice endpoint.
+ * The library will invoke this callback for each message arriving to the
+ * specified subject.
+ *
+ * @param req The request object, containing the message and other relevant
+ * references.
+ *
+ * @see microEndpointConfig, micro_endpoint_config_s.
+ */
+typedef microError *(*microRequestHandler)(microRequest *req);
+
+/** 
+ * @brief Callback type for async error notifications.
+ *
+ * If specified in microServiceConfig, this callback is invoked for internal
+ * errors (e.g. message delivery failures) related to a microservice. If the
+ * error is associated with an endpoint, the ep parameter points at the
+ * endpoint. However, this handler may be invoked for errors happening in
+ * monitoring subjects, in which case ep is NULL.
+ *
+ * The error handler is invoked asynchronously, in a separate theread.
+ *
+ * The error handler is not invoked for microservice-level errors that are sent
+ * back to the client as responses. Note that the error counts in
+ * microEndpointStats include both internal and service-level errors.
+ *
+ * @param m The microservice object.
+ * @param ep The endpoint object, or NULL if the error is not associated with an
+ * endpoint.
+ * @param s The NATS status for the error.
+ *
+ * @see microServiceConfig, micro_service_config_s.
+ */
+typedef void (*microErrorHandler)(microService *m, microEndpoint *ep, natsStatus s);
+
+/** @} */ // end of microCallbacks
+
+/** \defgroup microStructs Public structs
+ *
+ *  Microservice public structs.
+ * 
+ *  @{
+ */
+
+/**
+ * The Microservice endpoint configuration object.
+ */
+struct micro_endpoint_config_s
+{
+    /**
+     * @brief The name of the endpoint.
+     *
+     * Used in the service stats to list endpoints by name. Must not be empty.
+     */
+    const char *Name;
+
+    /**
+     * @brief The NATS subject the endpoint will listen on.
+     *
+     * Wildcards are allowed. If `Subject` is empty, it attempts to default to
+     * `Name`, provided it is a valid subject.
+     *
+     * For endpoints added to a group, the subject is automatically prefixed
+     * with the group's prefix.
+     */
+    const char *Subject;
+
+    /**
+     * @brief The endpoint schema.
+     */
+    microSchema *Schema;
+
+    /**
+     * @brief The request handler for the endpoint.
+     */
+    microRequestHandler Handler;
+
+    /**
+     * @brief A user-provided pointer to store with the endpoint
+     * (state/closure).
+     */
+    void *State;
+};
+
+/**
+ * The Microservice endpoint stats struct.
+ */
+struct micro_endpoint_stats_s
+{
+    const char *Name;
+    const char *Subject;
+    
+    /**
+     * @brief The number of requests received by the endpoint.
+     */
+    int64_t num_requests;
```

Comment:
As discussed on a call, I originally - to be more like Go when it came to public structure fields - used camel case and upper case for the first letter. You did so for the Name and Subject above but kept the ones below as `num_requests`, which is, I agree, more C like, but a bit of a departure from the rest. I would not mind that you have all new micro structs/fields be different from the rest, but then I would say that you did not need to change `Name`, `Subject` and other fields casing. Let's try to be consistent, especially for the same struct.

Reply:
Oops, I fixed most but obviously missed some, will fix.

Reply:
done